src/include/liburing.h

```diff
@@ -459,7 +459,7 @@ static inline void io_uring_prep_shutdown(struct io_uring_sqe *sqe, int fd,
 static inline unsigned io_uring_sq_ready(struct io_uring *ring)
 {
 	/* always use real head, to avoid losing sync for short submit */
-	return ring->sq.sqe_tail - *ring->sq.khead;
+	return ring->sq.sqe_tail - io_uring_smp_load_acquire(ring->sq.khead);
```

Comment:
Since this is only strictly needed for SQPOLL, maybe we should make it dependent on that? Something ala:

```
if (!(ring->flags & IORING_SETUP_SQPOLL))
    return ring->sq.sqe_tail - io_uring_smp_load_acquire(ring->sq.khead);
return ring->sq.sqe_tail - *ring->sq.khead;
```

Totally untested, just typed it in here... Then probably also include a comment as to why.

BTW, please include a Signed-off-by: Real Name <email> at the end of your commit. Thanks!

Reply:
Thanks, done. The bitwise condition was fine but needed to be flipped around.